src/mods/storytimer.cpp

```diff
@@ -0,0 +1,490 @@
+#include "storytimer.h"
+
+#include "mkb/mkb.h"
+
+#include "mods/freecam.h"
+#include "systems/assembly.h"
+#include "systems/pad.h"
+#include "systems/pref.h"
+#include "utils/draw.h"
+#include "utils/patch.h"
+#include "utils/timerdisp.h"
+
+namespace storytimer {
+
+enum class TimerOptions {
+    DontShow = 0,
+    AlwaysShow = 1,
+    BetweenWorlds = 2,
+    EndOfRun = 3,
+};
+
+static u32 s_spin_in_timer;
+static u32 s_gameplay_timer;
+static u32 s_postgoal_timer;
+static u32 s_stage_select_timer;
+static u32 s_exit_game_timer;
+static u32 s_fallout_timer;
+static u32 s_timeover_timer;
+static u32 s_spin_in_timer_correction;
+static u32 s_game_scenario_return_timer_correction;
+static u32 s_world_start_timer_correction;
+static u32 s_loadless_story_timer;
+static bool s_in_story;
+static bool s_is_on_world[11];
+static bool s_is_between_worlds;
+static bool s_is_run_complete;
+static bool s_is_on_spin_in;
+static bool s_is_on_gameplay;
+static bool s_is_postgoal;
+static bool s_is_on_stage_select_screen;
+static bool s_is_on_exit_game_screen;
+static bool s_is_on_fallout_screen;
+static bool s_is_timeover;
+static bool s_can_increment_stage_counter;
+static bool s_lower_stage_counter;
+static bool s_start_STAGE_FADE_OUT_TIMEr;
```

Comment:
Vast majority of these don't need to be static variables (or perhaps not even variables at all)

Reply:
I made most of these local variables, if you think most don't need to be variables at all I can do that. My reasoning for keeping them as variables is that (I hope) writing it this way makes it easier to read than having like 20 submodes in an if()